src/d3a/setup/strategy_tests/external_devices.py

```diff
@@ -0,0 +1,71 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+from d3a.models.appliance.switchable import SwitchableAppliance
+from d3a.models.area import Area
+from d3a.models.strategy.storage import StorageStrategy
+from d3a.models.strategy.load_hours import CellTowerLoadHoursStrategy, LoadHoursStrategy
+from d3a.models.appliance.pv import PVAppliance
+from d3a.models.strategy.external_strategies.pv import PVExternalStrategy
+from d3a.models.strategy.external_strategies.load import LoadHoursExternalStrategy
+
+
+def get_setup(config):
+    area = Area(
+        'Grid',
+        [
+            Area(
+                'House 1',
+                [
+                    Area('H1 General Load', strategy=LoadHoursStrategy(avg_power_W=200,
+                                                                       hrs_per_day=6,
+                                                                       hrs_of_day=list(
+                                                                           range(12, 18)),
+                                                                       final_buying_rate=35),
+                         appliance=SwitchableAppliance()),
+                    Area('H1 Storage1', strategy=StorageStrategy(initial_soc=50),
+                         appliance=SwitchableAppliance()),
+                    Area('H1 Storage2', strategy=StorageStrategy(initial_soc=50),
+                         appliance=SwitchableAppliance()),
+                ],
+                transfer_fee_pct=0, transfer_fee_const=0,
+            ),
+            Area(
+                'House 2',
+                [
+                    Area('load', strategy=LoadHoursExternalStrategy(avg_power_W=200,
+                                                                    hrs_per_day=4,
+                                                                    hrs_of_day=list(
+                                                                        range(12, 16)),
+                                                                    final_buying_rate=35),
```

Comment:
Shouldn't it be throwing an exception when user wants to use `LoadHoursExternalStrategy` and simultaneously also tries to set price preference?

Reply:
That's again the confusing requirement we discussed in the comments above. We need to be able to set the price preference for the default strategies, the external strategy only takes over after a client is connected/registered.

Reply:
Ohkay, I think this needs to be added in test somehow, as I fear strategy placing default offer/bid in addition to external agent's offer/bid, leading to double counting

Reply:
I think there is a misunderstanding here. This happens in either/or fashion. Either the default strategy is used or the external agent. There can never be a case where the default strategy will place offers at the same time as the external agent.